### ISSUE TITLE
Deprecate a bunch of figure factories

### ISSUE THREAD

**Issue by nicolaskruchten** [plotly/plotly.py#1922]
These figure factories should probably be marked "deprecated" in the docstrings:

These have dedicated trace types now:

```
create_bullet(data[, markers, measures, …])
create_ohlc(open, high, low, close[, dates, …])
create_scatterplotmatrix(df[, index, …])
create_table(table_text[, colorscale, …])
create_violin(data[, data_header, …])
create_candlestick(open, high, low, close[, …])
```

These are basically best done using PX if possible:

```
create_2d_density(x, y[, colorscale, …])
create_distplot(hist_data, group_labels[, …])
create_facet_grid(df[, x, y, facet_row, …])
```

**Comment by emmanuelle** [comment]
Should I just modifiy the docstrings or should we completely deprecate the functions?

**Comment by nicolaskruchten** [comment]
Let's start by just modifying the docstrings... Right now some of them say "BETA" which is kind of the wrong impression IMO :P

**Comment by emmanuelle** [comment]
Addressed in #1921.
